lib/model/query/entities.js

```diff
@@ -144,11 +144,24 @@ const processSubmissionEvent = (event) => (container) =>
       // The JSON.stringify appears to strip out error.stack so that doesn't make it to the
       // log details even for our Problems.
       container.Audits.log({ id: event.actorId }, 'entity.create.error', null,
-        { submissionId: event.details.submissionId,
-          submissionDefId: event.details.submissionDefId,
+        { submissionId,
+          submissionDefId,
           errorMessage: err.message,
           problem: (err.isProblem === true) ? err : null }));
 
+const processSubmissionEvent = (event) =>
+  _processionSubmissionEvent(event, event.details.submissionId, event.details.submissionDefId, event.acteeId);
+
+const createEntitiesFromPendingSubmissions = (pendingSubmissions, event) => (container) =>
```

Comment:
eventId for logging should be submission create event or latest update event

Reply:
and add information about this grand event in details column of entity_source

Reply:
Just my two cents, I vote for making the `dataset.update` event the primary source event. That's because the `dataset.update` event is the event in the causal chain that immediately preceded the creation of the entity. Sort of like how with submission approval, the approval is just the last step that caused the creation of the entity. The entity &hellip;/audits endpoint should still return the `submissionCreate` property, but it seems potentially helpful for the `sourceEvent` returned to be the `dataset.update` event. No strong feelings, feel free to ignore!

Reply:
will you show this event on the frontend?